src/uu/ls/src/ls.rs

```diff
@@ -2037,12 +2037,22 @@ impl PathData {
 }
 
 fn show_dir_name(path_data: &PathData, out: &mut BufWriter<Stdout>, config: &Config) {
-    if config.hyperlink && !config.dired {
+    let (name, has_colon) = if config.hyperlink && !config.dired {
         let name = escape_name(path_data.p_buf.as_os_str(), &config.quoting_style);
         let hyperlink = create_hyperlink(&name, path_data);
-        write!(out, "{}:", hyperlink).unwrap();
+        (hyperlink, name.contains(':'))
```

Comment:
What's so special about `:`? Shouldn't this be handled inside `escape_name`? Are there other special characters that behave weirdly?

Reply:
It appears that this solution indeed only works for the default quoting style in TTY (Shell), and might not for other quoting styles.

However, the `escape_name` function shall not **always** escape the colon `:` character, as shown in this example : 

```
➜  test git:(renji/fix-6554) ✗ ls --quoting-style=shell  
dir:name  dirname2  x
➜  test git:(renji/fix-6554) ✗ ls --quoting-style=shell -R
.:
dir:name  dirname2  x

'./dir:name':

./dirname2:
```

Here, the `dir:name` printed as an element from the current directory is not escaped, while the title of the directory is.

I will take care of fixing this

Reply:
Huh, that's weird behavior.

Reply:
Yes, and I have found another weird case with the space escaping, in `escape` quoting style. The space does not get escaped in the `dirname` mode, but it does in the regular display mode.

```
$ ls -R --quoting-style=escape
.:
dir\ name3  dir$name5  x

./dir name3:

./dir$name5:
```

Reply:
I have tested the behavior with some other characters (namely `[' ', '$', '\'', ':', '\\']`) and checked their test output.
You can take a look at it [here](https://github.com/uutils/coreutils/compare/b3b85048890dc9fed87c4fcc574512c42ac6c6ee..f12114915a05c58854315cbf8bd68c669116938f#diff-59ee1b49747d8d313fbe328a3f3f0997f9917c8934c7442db705288470a4007aR2207-R2325)